rpki-validator/src/main/java/net/ripe/rpki/validator3/util/ForkJoin.java

```diff
@@ -0,0 +1,38 @@
+package net.ripe.rpki.validator3.util;
+
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+public class ForkJoin {
+    /**
+     * Indicates that supplier is potentially blocking so {@link ForkJoinPool} can spawn additional threads if needed.
+     *
+     * @param supplier potentially blocking operation
+     * @param <T>      type of result
+     * @return result of Supplier#get
+     * @see ForkJoinPool#managedBlock
+     */
+    public static <T> T blocking(Supplier<T> supplier) {
+        AtomicReference<T> result = new AtomicReference<>();
+        try {
+            ForkJoinPool.managedBlock(new ForkJoinPool.ManagedBlocker() {
+                @Override
+                public boolean block() {
+                    result.set(supplier.get());
+                    return true;
+                }
+
+                @Override
+                public boolean isReleasable() {
+                    return false;
```

Comment:
Blocking queue example I found [here](https://www.concretepage.com/java/jdk7/example-of-managedblocker-in-java) gives me impression that maybe here we can release if the result is set?

Reply:
The fork-join pool will release the thread as soon as one of `isReleasable` or `block` returns true. So in our case it will call `isReleasable`,  get `false` so call `block`, then get `true` and then we're done.